app/championship.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: list[dict],
+            weapon: dict,
+            potion: dict | None
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.knight_statistics = {
+            "hp": None,
+            "power": None,
+            "protection": None
+        }
+
+    def get_knight_statistics(self) -> dict:
+        hp = self.hp
+        power = self.power + self.weapon["power"]
+        protection = 0
+
+        for armour_type in self.armour:
+            protection += armour_type["protection"]
+
+        if self.potion is not None:
+            effect = self.potion.get("effect")
+            power += effect.get("power", 0)
+            protection += effect.get("protection", 0)
+            hp += effect.get("hp", 0)
+
+        return {
+            "hp": hp,
+            "power": power,
+            "protection": protection
+        }
+
+
+class Battle:
+    def __init__(self, first_fighter: Knight, second_fighter: Knight) -> None:
+        self.first_fighter = first_fighter
+        self.second_fighter = second_fighter
+        self.result = {}
+
+    @staticmethod
+    def normalize_hp(hp: int) -> int:
+        if hp < 0:
+            return 0
+
+        return hp
+
+    def battle(self) -> None:
+        f_fighter = self.first_fighter.get_knight_statistics()
+        s_fighter = self.second_fighter.get_knight_statistics()
+
+        first_fighter_hp = (
+            f_fighter["hp"] + f_fighter["protection"] - s_fighter["power"]
+        )
+        second_fighter_hp = (
+            s_fighter["hp"] + s_fighter["protection"] - f_fighter["power"]
+        )
+
+        self.result = {

```

Comment:
Just use return and remove result from instance

Reply:
Done

Reply:
Not fixed

Reply:
Sorry